lib/config/configuration.go

```diff
@@ -1111,6 +1111,14 @@ func applyProxyConfig(fc *FileConfig, cfg *servicecfg.Config) error {
 		cfg.Proxy.MySQLServerVersion = fc.Proxy.MySQLServerVersion
 	}
 
+	if fc.Proxy.AutomaticUpgradesChannels != nil {
+		cfg.Proxy.AutomaticUpgradesChannels = fc.Proxy.AutomaticUpgradesChannels
+		err = cfg.Proxy.AutomaticUpgradesChannels.CheckAndSetDefaults()
+		if err != nil {
+			return trace.Wrap(err, "failed to set up automatic update channels")
+		}
```

Comment:
:nail_care: 
Reduce err scope
```suggestion
		if err := cfg.Proxy.AutomaticUpgradesChannels.CheckAndSetDefaults(); err != nil {
			return trace.Wrap(err, "failed to set up automatic update channels")
		}
```